pyembed/src/interpreter.rs

```diff
@@ -405,19 +427,12 @@ impl<'python, 'interpreter, 'resources> MainPythonInterpreter<'python, 'interpre
     /// instance. This is because `MainPythonInterpreter.drop()` finalizes
     /// the interpreter. The borrow checker should refuse to compile code
     /// where the returned `Python` outlives `self`.
-    pub fn acquire_gil(&mut self) -> cpython::Python<'_> {
-        match self.py {
-            Some(py) => py,
-            None => {
-                let gil = cpython::GILGuard::acquire();
-                let py = unsafe { cpython::Python::assume_gil_acquired() };
-
-                self._gil = Some(gil);
-                self.py = Some(py);
-
-                py
-            }
+    pub fn acquire_gil(&mut self) -> Python<'_> {
```

Comment:
I think the implementation of this function can just be `self.gil.get_or_insert_with(Python::acquire_gil).python()`